File: cpp/3392_count_subarrays/count_subarrays.cpp
#include <cassert>
#include <tuple>
#include <vector>

using namespace std;

class Solution {
 public:
  int countSubarrays(vector<int>& nums) {
    int n = nums.size();
    int ans = 0;
    for (int i = 1; i < n - 1; ++i) {
      if (nums[i] == (nums[i - 1] + nums[i + 1]) * 2) {
        ++ans;
      }
    }
    return ans;
  }
};

int main() {
  vector<tuple<vector<int>, int>> tests{
      {{1, 2, 1, 4, 1}, 1},
      {{1, 1, 1}, 0},
  };

  for (auto& [nums, ans] : tests) {
    assert(Solution().countSubarrays(nums) == ans);
  }
}

File: rust/3392_count_subarrays/count_subarrays.rs
struct Solution;

impl Solution {
    pub fn count_subarrays(nums: Vec<i32>) -> i32 {
        let n = nums.len();
        let mut ans = 0;
        for i in 1..n - 1 {
            if nums[i] == (nums[i - 1] + nums[i + 1]) * 2 {
                ans += 1;
            }
        }
        ans
    }
}

fn main() {
    let tests = vec![(vec![1, 2, 1, 4, 1], 1), (vec![1, 1, 1], 0)];

    for (nums, ans) in tests {
        assert_eq!(Solution::count_subarrays(nums), ans);
    }
}
